### PR TITLE
Project 6: Views

### PR DESCRIPTION
**Pull request by MoRaad97**
## Project requirements
- Implement the design from the sneak peek wireframes.
    - Styling is nice to have but focus on making sure everything functions first. (Note: Ignore login logic, you would be using Devise in coming milestone)
    
- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).

     #### [....More Details](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Views_project.md)

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,21 @@
...
+  </thead>
+  <tbody>
+    <% @users.each do |user| %>
+    <tr>
+      <td>
+        <img src="#{<%= user.photo %>}" alt="User Img" />
+      </td>
+      <td><%= user.name %></td>
+      <td><%= user.posts_counter %></td>
+    </tr>
```
Comment (omarsalem7) [inline app/views/users/index.html.erb]: - [x] Kindly add a link for each user when you click on it suppose takes you to their show page as shown in the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_page.png)

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,25 @@
...
+  <span><%= @user.posts_counter %></span>
+</div>
+<div>
+<h2>Bio</h2>
+  <p> <%= @user.bio || "the bio is embty" %></p>
+</div>
+<div>
+<h2>Posts</h2>
+<ol>
+<% @posts.each do |post| %>
```
Comment (omarsalem7) [inline app/views/users/show.html.erb:15]: - [x] Consider in the show view to show the recent 3 posts, not all posts by creating a method for the recent 3 posts.

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,25 @@
...
+<li>
+<p><%= post.title%></p>
+<p><%= post.text%></p>
+<p>number on comments:<%= post.comment_counter || 0%></p>
+<p>number on likes:<%= post.likes_counter || 0%></p>
+</li>
+<% end %>
+</ol>
```
Comment (omarsalem7) [inline app/views/users/show.html.erb:23]: - [x] Also consider adding the **See all Post** link which opens the posts index view as displayed on the [wireframe](https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_user_all_posts.png)

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,21 @@
...
+<div>
+<h2>Posts</h2>
+<ol>
+<% @posts.each do |post| %>
+<li>
+<p><%= post.title%></p>
+<p><%= post.text%></p>
+<p>number on comments:<%= post.comment_counter || 0%></p>
+<p>number on likes:<%= post.likes_counter || 0%></p>
+</li>
```
Comment (omarsalem7) [inline app/views/posts/index.html.erb:17]: - [x] Kindly use `post.recent_comments` to render 5 most recent comments as in this [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/blog-app/images/blog_user_all_posts.png?token=GHSAT0AAAAAABYH576Q6LJTZRQ6P5CMNUC2YZUPHSA)

**Review by omarsalem7 — APPROVED** [review]
Hi @MoRaad97 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (MoRaad97) [reply app/views/posts/index.html.erb:17]: Thanks.

File: app/views/users/index.html.erb (hunk 1 above)
Comment (ShoiraTa) [reply app/views/users/index.html.erb]: Link is added 👍🏻

File: app/views/users/show.html.erb (hunk 2 above)
Comment (ShoiraTa) [reply app/views/users/show.html.erb:15]: Implemented 👍🏻

File: app/views/users/show.html.erb (hunk 3 above)
Comment (ShoiraTa) [reply app/views/users/show.html.erb:23]: Added 👍🏻

File: app/views/posts/index.html.erb (hunk 4 above)
Comment (ShoiraTa) [reply app/views/posts/index.html.erb:17]: Added 👍🏻

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @MoRaad97 , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

### Highlights:
 ✔️ Project is setup correctly
 ✔️ All linters are passing
 ✔️ Correct gitflow
 ✔️ Professional documentation

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

image

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@shoirata] in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**